tests/test_utils_strict_dataclass.py

```diff
@@ -0,0 +1,575 @@
+        # Any
+        (5, Any),
+        ("John", Any),
+        (DummyClass(), Any),
+        # Deep nested type
+        (
+            {
+                "a": [
+                    (1, DummyClass(), {1, "2", "3", 4}),
+                    (2, DummyClass(), None),
+                ],
+            },
+            Dict[
+                str,
+                List[
+                    Tuple[
+                        int,
+                        DummyClass,
+                        Optional[Set[Union[int, str],]],
+                    ]
+                ],
+            ],
+        ),
+    ],
+)
+def test_type_validator_valid(value, type_annotation):
+    type_validator("dummy", value, type_annotation)
+
+
+@pytest.mark.parametrize(
+    "value, type_annotation",
+    [
+        # Basic types
+        (5, float),
+        (5.0, int),
+        ("John", int),
+        # Union types
+        (5.0, Union[int, str]),
+        (None, Union[int, str]),
+        (DummyClass(), Union[int, str]),
+        # Optional
+        ("John", Optional[int]),
+        (DummyClass(), Optional[int]),
+        # Literal
+        ("Ada", Literal["John", "Doe"]),
+        (3, Literal[4, 5, 6]),
+        # List
+        (5, List[int]),
+        ([1, 2, "3"], List[int]),
+        # Tuple
+        (5, Tuple[int, int, int]),
+        ((1, 2, "3"), Tuple[int, int, int]),
+        ((1, 2, 3, 4), Tuple[int, int, int]),
+        ((1, 2, "3", 4), Tuple[int, ...]),
+        # Dict
+        (5, Dict[str, int]),
+        ({"a": 1, "b": "2"}, Dict[str, int]),
+        # Set
+        (5, Set[int]),
+        ({1, 2, "3"}, Set[int]),
+        # Custom classes
+        (5, DummyClass),
+        ("John", DummyClass),
+    ],
+)
+def test_type_validator_invalid(value, type_annotation):
+    with pytest.raises(TypeError):
+        type_validator("dummy", value, type_annotation)
+
+
+class DummyValidator:
+    def __init__(self, threshold):
+        self.threshold = threshold
+
+    def __call__(self, value):
+        return value < self.threshold
+
+    def compare(self, value, value2=10):
+        return value < value2
+
+
+@pytest.mark.parametrize(
+    "obj",
+    [
+        positive_int,
+        multiple_of_64,
+        lambda value: None,
+        lambda value, factor=2: None,
+        lambda value=1, factor=2: value * factor,
+        lambda *values: None,
+        DummyValidator(threshold=10),  # callable object
+        DummyValidator(threshold=10).compare,  # callable method
+    ],
+)
+def test_is_validator(obj):
+    # Anything that can be called with `obj(value)` is a correct validator.
+    assert _is_validator(obj)
+
+
+@pytest.mark.parametrize(
+    "obj",
+    [
+        positive_int,
+        multiple_of_64,
+        lambda: None,  # no argument
+        lambda value1, value2: None,  # more than one argument with default values
+        lambda *, value: None,  # keyword-only argument
+    ],
+)
+def test_not_a_validator(obj):
+    assert not _is_validator(5)
```

Comment:
```suggestion
    assert not _is_validator(obj)
```
and drop `positive_int` and `multiple_of_64` in the parametrize

Reply:
well well well... :see_no_evil: